frontend/src/components/shared/ComparisonMatrix.vue

```diff
@@ -0,0 +1,201 @@
+<template>
+  <div v-if="mergedComparisons" class="table-container">
+    <table class="table">
+      <thead>
+        <tr>
+          <th></th>
+          <th v-for="cn in columnNames" :key="cn">{{ cn }}</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr v-for="(rn, i) in rowNames" :key="rn + i">
+          <th>{{ rn }}</th>
+          <td v-for="(cn, j) in columnNames" :key="cn + j" :class="{selected: isSelectedCell(i, j)}">
+            <div>
+              <span v-for="(type, k) in types" :key="cn + j + type"
+                    :style="{backgroundColor: colors[k]}"
+                    @click="handleSelectCell(i, j)">
+                {{ matrix[i][j][type] }}
+              </span>
+            </div>
+          </td>
+        </tr>
+      </tbody>
+      <caption>
+        click a cell to see the comparison details
+        <br />
+        legend:
+        <span v-for="(type, k) of types" :key="type" :style="{backgroundColor: colors[k]}">
+          {{ type }}
+        </span>
+      </caption>
+    </table>
+  </div>
+</template>
+
+<script>
+
+import { mapState } from 'vuex';
+
+export default {
+  name: 'ComparisonMatrix',
+  data() {
+    return {
+      colors: ['#eef6fc', '#fffbeb'],
+    };
+  },
+  computed: {
+    ...mapState({
+      comparisons: state => state.compare.comparisons,
+      modelList: state => state.models.modelList,
+      selectedCell: state => state.compare.selectedCell,
+    }),
+    mergedComparisons() {
```

Comment:
Just a note: in theory it would make sense to compute this in the `api` so that it would be cached, instead of having each client do this computation.